src/workflows-app/SubmissionHistory.test.js

```diff
@@ -0,0 +1,561 @@
+const originalOffsetHeight = Object.getOwnPropertyDescriptor(HTMLElement.prototype, 'offsetHeight');
+const originalOffsetWidth = Object.getOwnPropertyDescriptor(HTMLElement.prototype, 'offsetWidth');
+
+const runSetData = {
+  run_sets: [
+    {
+      error_count: 0,
+      submission_timestamp: '2022-01-01T12:00:00.000+00:00',
+      last_modified_timestamp: '2022-01-02T13:01:01.000+00:00',
+      record_type: 'FOO',
+      run_count: 1,
+      run_set_id: 'ea001565-1cd6-4e43-b446-932ac1918081',
+      state: 'COMPLETE',
+    },
+    {
+      error_count: 1,
+      submission_timestamp: '2021-07-10T12:00:00.000+00:00',
+      last_modified_timestamp: '2021-08-11T13:01:01.000+00:00',
+      record_type: 'FOO',
+      run_count: 2,
+      run_set_id: 'b7234aae-6f43-405e-bb3a-71f924e09825',
+      state: 'ERROR',
+    },
+  ],
+};
+
+beforeAll(() => {
+  Object.defineProperty(HTMLElement.prototype, 'offsetHeight', { configurable: true, value: 1000 });
+  Object.defineProperty(HTMLElement.prototype, 'offsetWidth', { configurable: true, value: 800 });
+});
+
+const cbasUrlRoot = 'https://lz-abc/terra-app-abc/cbas';
+const cromwellUrlRoot = 'https://lz-abc/terra-app-abc/cromwell';
+
+beforeEach(() => {
+  getConfig.mockReturnValue({ cbasUrlRoot, cromwellUrlRoot });
+  MenuTrigger.mockImplementation(({ content }) => {
+    return div({ role: 'menu' }, [content]);
+  });
+});
+
+afterEach(() => {
+  jest.clearAllMocks();
+});
+
+afterAll(() => {
+  Object.defineProperty(HTMLElement.prototype, 'offsetHeight', originalOffsetHeight);
+  Object.defineProperty(HTMLElement.prototype, 'offsetWidth', originalOffsetWidth);
+});
+
+// Note: Since the timestamps in the data is being converted to Local timezone, it returns different time when the tests
+//       are run locally and in GitHub action. Hence everywhere in this file we are verifying only the date format for now.
+describe('SubmissionHistory page', () => {
+  const headerPosition = {
+    Actions: 0,
+    Submission: 1,
+    Status: 2,
+    'Date Submitted': 3,
+    Duration: 4,
+    Comment: 5,
+  };
+
+  it('should sort columns properly', async () => {
+    const user = userEvent.setup();
+    const getRunSetsMethod = jest.fn(() => Promise.resolve(runSetData));
+    const mockLeoResponse = jest.fn(() => Promise.resolve(mockAzureApps));
+
+    await Ajax.mockImplementation(() => {
+      return {
+        Cbas: {
+          runSets: {
+            get: getRunSetsMethod,
+          },
+        },
+        Apps: {
+          listAppsV2: mockLeoResponse,
+        },
+      };
+    });
+
+    // Act
+    await act(async () => {
+      await render(
+        h(BaseSubmissionHistory, {
+          name: 'test-azure-ws-name',
+          namespace: 'test-azure-ws-namespace',
+          workspace: mockAzureWorkspace,
+        })
+      );
+    });
+
+    await waitFor(() => {
+      expect(screen.getByRole('table')).toBeInTheDocument();
+    });
+
+    const table = await screen.getByRole('table');
```

Comment:
`getBy...` queries are synchronous and do not need to be awaited.
https://testing-library.com/docs/queries/about#types-of-queries